### PR TITLE
Modal Event-Triggered Actions

### PR DESCRIPTION
**Pull request by DXCanas**
## Summary

Light refactor, clears data on exit using the modal's x button. Clears data using props passed in.

## Issues addressed

- https://trello.com/c/VyhuoydF/435-password-reset-form-submitting-with-both-fields-blank-does-not-show-any-error
- https://trello.com/c/oLd4nPrW/430-opening-an-edit-user-modal-returns-to-previous-state

### CODE REVIEW

**Comment by DXCanas** [comment]
@jtamiace If everything looks good, we can get this in master before we build

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-    <modal v-ref:modal btntext="Edit">
+    <modal v-on:open="clear()" v-ref:modal btntext="Edit">
 
       <h1 slot="header" class="header">Edit Account Info</h1>
 
-      <div @keyup.enter="editUser" v-if="!usr_delete && !pw_reset" slot="body">
+      <div @keyup.enter="editUser" v-if="!this.usr_delete && !this.pw_reset" slot="body">
 
         <div class="user-field">
           <label for="username">Full Name</label>:
```
Comment (radinamatic) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue]: Label mixup here? `input id="name"` but `label for="username"`
@DXCanas @indirectlylit 

Check the lines 16 & 17 too!

**Comment by DXCanas** [comment]
@radinamatic updated

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue [hunk 2]
```diff
@@ -1,19 +1,19 @@
 <template>
 
   <div class="user-edit-modal">
-    <modal v-ref:modal btntext="Edit">
+    <modal v-on:open="clear()" v-ref:modal btntext="Edit">
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue:4]: in the future, let's stick with the `@open` shorthand

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
 
       <h1 slot="header" class="header">Edit Account Info</h1>
 
-      <div @keyup.enter="editUser" v-if="!usr_delete && !pw_reset" slot="body">
+      <div @keyup.enter="editUser" v-if="!this.usr_delete && !this.pw_reset" slot="body">
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue:8]: is it correct to have `this` inside the template?

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue [hunk 4]
```diff
@@ -25,7 +25,7 @@
           </select>
         </div>
 
-        <div class="advanced-options" v-if="!pw_reset && !usr_delete">
+          <div class="advanced-options" v-if="!this.usr_delete && !this.pw_reset">
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue:28]: indentation

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue (hunk 3 above)
Comment (rtibbles) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue:8]: Should be redundant.

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue (hunk 2 above)
Comment (DXCanas) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue:4]: wasn't sure if `@open` was a thing. I saw `@click` everywhere and assumed @ notation was only for HTML events

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue (hunk 3 above)
Comment (DXCanas) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue:8]: woops

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue (hunk 4 above)
Comment (DXCanas) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue:28]: surprised the linter didn't get this

**Pull request merged**